src/org/nejrasm/zadaca3oop/task2/ElectricCar.java

```diff
@@ -0,0 +1,33 @@
+package org.nejrasm.zadaca3oop.task2;
+
+public class ElectricCar extends Vehicle {
+    private final ElectricEngine electricCarEngine;
+
+    public ElectricCar(final String model, final double capacity) {
+        super(model);
+        this.electricCarEngine = new ElectricEngine(capacity);
+    }
+
+    @Override
+    public double accelerate(final double acceleration) {
+        final double capacityState = this.electricCarEngine.decreaseCapacity(0.4 * acceleration / 100);
+        if ((this.electricCarEngine.capacity - capacityState) > 0) {
+            this.electricCarEngine.capacity = capacityState;
```

Comment:
You made `electricCarEngine.capacity` `protected` to be able to handle capacity management in the car.
And kind of abused the fact that protected fields are accessible in the same package.

The car should not manage the capacity, that should be done by the engine. That way `capacity` could have stayed `private`.

Reply:
Fixed. I didn't understood well this part earlier. Thanks :)